### PR TITLE
[TMP] Add events to have better logger and possible outputs

### PR DESCRIPTION
**Pull request by pierallard**
This is a temporary pull request, can be full of todo's or weird stuff.

### CODE REVIEW

File: spec/Akeneo/Crowdin/PackagesDownloaderSpec.php [hunk 1]
```diff
@@ -7,12 +7,13 @@
 use Akeneo\Crowdin\Client;
 use PhpSpec\ObjectBehavior;
 use Prophecy\Argument;
+use Symfony\Component\EventDispatcher\EventDispatcher;
 
 class PackagesDownloaderSpec extends ObjectBehavior
 {
-    function let(Client $client)
+    function let(Client $client, EventDispatcher $eventDispatcher)
```
Comment (jjanvier) [inline spec/Akeneo/Crowdin/PackagesDownloaderSpec.php]: you should use interfaces everywhere for such objects ;à

File: src/Akeneo/Crowdin/PackagesDownloader.php [hunk 2]
```diff
@@ -17,12 +20,17 @@ class PackagesDownloader
...
+    /** @var EventDispatcher */
+    protected $eventDispatcher;
+
     /**
-     * @param Client $client
+     * @param Client          $client
+     * @param EventDispatcher $eventDispatcher
      */
-    public function __construct(Client $client)
+    public function __construct(Client $client, EventDispatcher $eventDispatcher)
```
Comment (jjanvier) [inline src/Akeneo/Crowdin/PackagesDownloader.php]: interface

File: src/Akeneo/System/AbstractConsoleLogger.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * This class is an event subscriber and contains methods to display messages on console.
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AbstractConsoleLogger implements EventSubscriberInterface
```
Comment (jjanvier) [inline src/Akeneo/System/AbstractConsoleLogger.php]: abstract class

File: src/Akeneo/System/AbstractConsoleLogger.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        $this->output = new ConsoleOutput();
+        $formatter = $this->output->getFormatter();
+        $formatter->setStyle('blink', new OutputFormatterStyle(null, null, array('blink')));
+        $formatter->setStyle('bold', new OutputFormatterStyle(null, null, array('bold')));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
```
Comment (jjanvier) [inline src/Akeneo/System/AbstractConsoleLogger.php]: to remove, should be implemented by the children

File: src/Akeneo/Crowdin/TranslationFilesCreator.php [hunk 5]
```diff
@@ -61,11 +69,18 @@ public function create(array $files, TranslationProjectInfo $projectInfo, $baseB
                     $file->getProjectDir(),
                     $file->getSource()
                 );
+
+                $this->eventDispatcher->dispatch(Events::CROWDIN_CREATE_FILE, new GenericEvent(null, [
```
Comment (nidup) [inline src/Akeneo/Crowdin/TranslationFilesCreator.php]: why "new GenericEvent(" with ```null``` why not passing ```$file``` ?

File: src/Akeneo/Crowdin/TranslationFilesUpdater.php [hunk 6]
```diff
@@ -59,10 +66,16 @@ public function update(array $files, $baseBranch)
                     $file->getProjectDir(),
                     $file->getSource()
                 );
+
+                $this->eventDispatcher->dispatch(Events::CROWDIN_UPDATE_FILE, new GenericEvent(null, [
```
Comment (nidup) [inline src/Akeneo/Crowdin/TranslationFilesUpdater.php]: why null?

File: src/Akeneo/Git/PullRequestCreator.php [hunk 7]
```diff
@@ -31,64 +38,72 @@ class PullRequestCreator
...
+                $this->repository,
+                [
+                    'head' => sprintf('%s:crowdin/%s', $this->fork_owner, $branch),
+                    'base' => $baseBranch,
+                    'title' => 'Update translations from Crowdin',
+                    'body' => 'Updated on ' . $branch,
+                ]
+            );
+        } catch (ValidationFailedException $exception) {
+            // This append when there is no modifications.
```
Comment (nidup) [inline src/Akeneo/Git/PullRequestCreator.php]: Means the PR will not be created?

I would raise a dedicated business event here to say that there is nothing to update and based on this event I would display an message "all translations are already up to date for the branch 1.x"

**Comment by nidup** [comment]
few comments, then GTM

**Pull request merged**